packages/tools/src/utilities/getCalibratedUnits.ts

```diff
@@ -56,7 +75,154 @@ const getCalibratedAreaUnits = (handles, image): string => {
  * Gets the scale divisor for converting from internal spacing to
  * image spacing for calibrated images.
  */
-const getCalibratedScale = (image) => image.calibration?.scale || 1;
+const getCalibratedScale = (image, handles = []) => {
+  if (image.calibration?.sequenceOfUltrasoundRegions) {
+    // image.spacing / image.us.space
+  } else if (image.calibration?.scale) {
+    return image.calibration.scale;
+  } else {
+    return 1;
+  }
+};
+
+/**
+ * Extracts the calibrated length units, area units, and the scale
+ * for converting from internal spacing to image spacing.
+ *
+ * @param handles - to detect if spacing information is different between points
+ * @param image - to extract the calibration from
+ * @returns Object containing the units, area units, and scale
+ */
+const getCalibratedLengthUnitsAndScale = (image, handles) => {
+  const [imageIndex1, imageIndex2] = handles;
+  const { calibration, hasPixelSpacing } = image;
+  let units = hasPixelSpacing ? 'mm' : PIXEL_UNITS;
+  const areaUnits = units + SQUARE;
+  let scale = 1;
+  let calibrationType = '';
+
+  if (
+    !calibration ||
+    (!calibration.type && !calibration.sequenceOfUltrasoundRegions)
+  ) {
+    return { units, areaUnits, scale };
+  }
+
+  if (calibration.type === CalibrationTypes.UNCALIBRATED) {
+    return { units: PIXEL_UNITS, areaUnits: PIXEL_UNITS + SQUARE, scale };
+  }
+
+  if (calibration.sequenceOfUltrasoundRegions) {
+    const supportedRegionsMetadata =
+      calibration.sequenceOfUltrasoundRegions.filter(
+        (region) =>
+          SUPPORTED_REGION_DATA_TYPES.includes(region.regionDataType) &&
+          SUPPORTED_LENGTH_VARIANT.includes(
+            `${region.physicalUnitXDirection},${region.physicalUnitYDirection}`
+          )
+      );
+
+    if (!supportedRegionsMetadata.length) {
+      return { units, areaUnits, scale };
+    }
+
+    const region = supportedRegionsMetadata.find(
+      (region) =>
+        imageIndex1[0] >= region.regionLocationMinX0 &&
+        imageIndex1[0] <= region.regionLocationMaxX1 &&
+        imageIndex1[1] >= region.regionLocationMinY0 &&
+        imageIndex1[1] <= region.regionLocationMaxY1 &&
+        imageIndex2[0] >= region.regionLocationMinX0 &&
+        imageIndex2[0] <= region.regionLocationMaxX1 &&
+        imageIndex2[1] >= region.regionLocationMinY0 &&
+        imageIndex2[1] <= region.regionLocationMaxY1
+    );
+
+    if (region) {
+      scale = 1 / (region.physicalDeltaX * region.physicalDeltaY * 100);
```

Comment:
If the scaling is not identical between the regions, then you get into some odd situations.  I think we should only allow the non-US specific tools to run if the units are the same between the two of them, and the physical delta is the same.
Otherwise, you could end up showing an invalid value for a measurement, which is a patient safety issue.